File: BinarySearch/BinarySearch.cpp
#include <bits/stdc++.h>
using namespace std;

int binarysearch(int arr[],int n,int key)
{
  int i=0,j=n-1,mid;
  while(i<=j)
  {
    mid=(i+j)/2;
    if(arr[mid]==key)
    {
      return mid;
    }
    else if(arr[mid]<key)
    {
      i=mid+1;
    }
    else
    {
      j=mid-1;
    }
  }
  return -1;
}
int main()
{
  int n,key,res_index;
  cout<<"enter the size of array: ";
  cin>>n;
  int arr[n];
  for(int i=0;i<n;i++)
  {
    cout<<"enter the element"<<i+1<<" : ";
    cin>>arr[i];
  }
  cout<<"enter the key to search: ";
  cin>>key;
  res_index=binarysearch(arr,n,key);
  if(res_index==-1)
  {
    cout<<"element "<<key<<" not in the list";
  }
  else
  {
    cout<<"element is present in index "<<res_index;
  }
  return 0;
}

File: BinarySearch/BinarySearch.py
def BinarySearch(arr,key):
     i,j=0,len(arr)-1
     while i<=j:
            mid=(i+j)//2
            if(arr[mid]==key):
                return mid
            elif(arr[mid]>key):
                j=mid-1
            else :i=mid+1
     return -1
    
def main():
    print("enter the arr: ")
    arr=[int(i) for i in input().split()]
    print("enter the key: ")
    key=int(input())
    if key!=-1:
        print("key element present at index "+str(BinarySearch(arr,key))+" in array")
    else:
        print("key is not present in array")
if __name__=="__main__":
    main()